WordPress/src/main/java/org/wordpress/android/ui/posts/PrepublishingViewModel.kt

```diff
@@ -1,13 +1,60 @@
 package org.wordpress.android.ui.posts
 
+import android.os.Bundle
+import android.os.Parcelable
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
+import kotlinx.android.parcel.Parcelize
+import org.wordpress.android.ui.posts.ActionState.TagsActionState
+import org.wordpress.android.ui.posts.PrepublishingActionItemUiState.ActionType
+import org.wordpress.android.ui.posts.PrepublishingScreen.HOME
+import org.wordpress.android.ui.posts.PrepublishingScreen.TAGS
+import org.wordpress.android.viewmodel.Event
 import javax.inject.Inject
 
+const val KEY_TAGS_ACTION_STATE = "key_tags_action_state"
+
+enum class PrepublishingScreen {
+    HOME,
+    PUBLISH,
+    VISIBILITY,
+    TAGS
+}
+
+@Parcelize
+data class ActionsState(val currentScreen: PrepublishingScreen = HOME, val tagsActionState: TagsActionState) :
+        Parcelable
+
+sealed class ActionState : Parcelable {
+    @Parcelize
+    data class TagsActionState(val tags: String?) : ActionState()
+}
+
 class PrepublishingViewModel @Inject constructor() : ViewModel() {
     private var isStarted = false
 
-    fun start() {
+    private lateinit var actionsState: ActionsState
+    private val _currentActionsState = MutableLiveData<Event<ActionsState>>()
+    val currentActionsState: LiveData<Event<ActionsState>> = _currentActionsState
+
+    fun start(actionsState: ActionsState) {
         if (isStarted) return
         isStarted = true
+        this.actionsState = actionsState
+        updateState()
+    }
+
+    private fun updateState() {
+        _currentActionsState.postValue(Event(actionsState))
+    }
+
+    fun updateCurrentActionTypeState(actionType: ActionType) {
+        actionsState = actionsState.copy(currentScreen = TAGS)
+        updateState()
+    }
+
+    fun writeToBundle(outState: Bundle) {
+        outState.putParcelable(KEY_TAGS_ACTION_STATE, actionsState)
     }
```

Comment:
This is a partial implementation, but the idea here is that every time there's an update from a screen it will modify the state here and that will get written when the state is being persisted.